encoding/tinycbor/src/cborparser.c

```diff
@@ -279,7 +279,9 @@ static CborError advance_internal(CborValue *it)
 {
     uint64_t length;
     CborError err = extract_number(it->parser, &it->offset,  &length);
-    assert(err == CborNoError);
+    if (err == CborNoError) {
+        return err;
+    }
```

Comment:
I was trying to follow the unroll of this error up to check that it is handled somewhere up in the call chain, it seems to be OK. But the same assert is also being checked inside `cbor_value_enter_container`, should not that one be fixed as well, for consistency?

Reply:
Yes. It should.

Reply:
hmm shouldn't if be !=  and not == ?